### PR TITLE
[AWS] Add support for Guardduty datastream

### LINKED ISSUE #2751: AWS GuardDuty
### Description
Amazon GuardDuty is a threat detection service that continuously monitors your AWS accounts and workloads for malicious activity and delivers detailed security findings for visibility and remediation. 

### Architecture
GuardDuty supports export of active findings to CloudWatch events, and optionally to an S3 bucket. More information on export options available [here](https://docs.aws.amazon.com/guardduty/latest/ug/guardduty_exportfindings.html). 

# Integration release checklist

This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

### All changes

- [ ] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md)
- [ ] Supported versions of the monitoring target are documented
- [ ] Supported operating systems are documented (if applicable)
- [ ] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [ ] Documentation exists
- [ ] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [ ] At least a manual test with ES / Kibana / Agent has been performed.
- [ ] Required Kibana version set to:

### New Package

- [ ] Screenshot of the "Add Integration" page on Fleet added

### Dashboards changes

- [ ] Dashboards exists
- [ ] Screenshots added or updated
- [ ] Datastream filters added to visualizations

### Log dataset changes

- [ ] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) exist (if applicable)
- [ ] Generated output for at least 1 log file exists
- [ ] Sample event (`sample_event.json`) exists

### PR DESCRIPTION
**Pull request by vinit-elastic**
## What does this PR do?

- Added a data stream.
- Added data collection logic to the data stream.
- Added the ingest pipeline for the data stream.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files
- Added dashboards and visualizations.
- Added test for pipeline for the data stream.
- Added system test cases for the data stream.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^8.4.0`).

## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/aws/data_stream directory.
- Run the following command to run tests. 
> elastic-package test 

## Related issues

- Closes #2751 

## Screenshots
image
image
image
image
image
image
image
image
image

### CODE REVIEW

File: packages/aws/data_stream/guardduty/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -0,0 +1,2482 @@
...
+      ignore_missing: true
+  - json:
+      field: event.original
+      target_field: json
+      on_failure:
+        - append:
+            field: error.message
+            value: '{{{_ingest.on_failure_message}}}'
+  - drop:
+      if: ctx.json?.findings != null && ctx.json.findings.isEmpty()
```
Comment (efd6) [inline packages/aws/data_stream/guardduty/elasticsearch/ingest_pipeline/default.yml:25]: Can we have a test for this situation?

File: packages/aws/data_stream/guardduty/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,2482 @@
+---
+description: Pipeline for processing Amazon GuardDuty Findings logs.
+processors:
+  - set:
+      field: ecs.version
+      value: '8.2.0'
```
[the remark this answers is not in the document]
Comment (vinit-elastic) [reply packages/aws/data_stream/guardduty/elasticsearch/ingest_pipeline/default.yml:6]: As we have ECS version 8.2.0 in the `build.yml` file. That's why we've kept the updated version of ECS.

File: packages/aws/changelog.yml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.29.0"
+  changes:
+    - description: Add Guardduty data stream.
```
[the remark this answers is not in the document]
Comment (vinit-elastic) [reply packages/aws/changelog.yml:4]: We will rephrase the description

**Comment by jamiehynds** [comment]
@efd6 is this ok to merge or any further changes required from Crest?

**Review by drewdaemon — APPROVED** [review]
Dashboards are following the key best practices 👍 

I see that you have used a non-Lens visualization: tag cloud. We will be adding tag clouds to Lens at some point but until then, we suggest using a Lens horizontal bar chart to convey the same information instead.

Of course, you can keep the legacy tag cloud here if you feel strongly. Just be aware that if you add non-Lens visualizations now, you will likely have to change them to Lens visualizations to use this dashboard on serverless.

**Review by efd6 — APPROVED** [review]
Thanks

**Comment by vinit-elastic** [comment]
> Dashboards are following the key best practices 👍
> 
> I see that you have used a non-Lens visualization: tag cloud. We will be adding tag clouds to Lens at some point but until then, we suggest using a Lens horizontal bar chart to convey the same information instead.
> 
> Of course, you can keep the legacy tag cloud here if you feel strongly. Just be aware that if you add non-Lens visualizations now, you will likely have to change them to Lens visualizations to use this dashboard on serverless.

Hey @drewdaemon,

Thank You for the suggestion,
Because the field may contain more **threat names**, I believe we will need to use tag clouds for these changes. If we use a bar chart, then it seems congested. 
What do you think? 🤔

**Comment by drewdaemon** [comment]
@vinit-elastic 

> The reason we opted for the tag cloud is that there could be multiple Threat Names and plotting it over a bar chart would result in clutter. If there are only a few values for threat names, I'd make sense to go for a bar chart in case of more than 10-15 names it would only show the first few names and others to be grouped under the "others" bucket and not grouping extra value would result in a visual mess.

Yes, I understand your dilemma.

My first question would be: how many threat names is it actually useful to show? Is it important to show more than the top 10-20 values? Generally, there's a law of diminishing returns at play in data visualization.

### Horizontal bar chart
The horizontal bar chart actually can support quite a few. As a comparable, here's an unrelated chart of about the same dimensions displaying 25 names (not necessarily the suggested number, just picking something big to illustrate).

Screenshot 2023-02-02 at 2 38 17 PM

It doesn't seem cluttered to me. The labels are readable, the bars are neatly aligned, and the user can mouse over any of them for more information. There is no "other bucket;" terms below the top 25 aren't shown.

A bar chart also makes the difference between one term and another much more obvious than does a tag cloud.

### Tag cloud

For example, here's a tag cloud of the same data.

Screenshot 2023-02-02 at 2 53 41 PM

Looking just at this tag cloud, I challenge you to tell me which is the more common term between `windows` and `pfsense`. This kind of a comparison is much easier with a bar chart.

That said, I understand the visual appeal of the tag cloud. They border on info-graphics. People like them for sure. 

Unfortunately, we're currently making you choose between using Lens and using a tag cloud. Ultimately, it's your job to do what's best for your users.

**Comment by vinit-elastic** [comment]
Hey @drewdaemon - Appreciate the explanation. It makes a whole lot of sense now. 
As you mentioned, a bar chart would be a better fit here. I'll update the visualisation from tag cloud to bar chart today. 
Let me know if you have anything else. 😄

**Review by endorama — APPROVED** [review]

**Pull request merged**